marlowe-chain-sync/src/Language/Marlowe/Runtime/ChainSync/NodeClient.hs

```diff
@@ -92,13 +96,14 @@ cost CostModel{..} Changes{..} = changesBlockCount * blockCost + changesTxCount
 
 -- | The set of dependencies needed by the NodeClient component.
 data NodeClientDependencies = NodeClientDependencies
-  { connectToLocalNode    :: !(LocalNodeClientProtocolsInMode CardanoMode -> IO ()) -- ^ Connect to the local node.
-  , getHeaderAtPoint      :: !(GetHeaderAtPoint IO)                                 -- ^ How to load a block header at a given point.
-  , getIntersectionPoints :: !(GetIntersectionPoints IO)                            -- ^ How to load the set of initial intersection points for the chain sync client.
+  -- Explicitly lazy to support value recursion
+  { connectToLocalNode    :: ~(LocalNodeClientProtocolsInMode CardanoMode -> IO ()) -- ^ Connect to the local node.
```

Comment:
Is the use of laziness here important for fixing the bug?

Reply:
Yes - NodeClient and ChainStore depend recursively on each other now via `ArrowLoop`.